go/vt/vttablet/tabletmanager/vreplication/vcopier_test.go

```diff
@@ -1397,6 +1398,88 @@ func testPlayerCopyTablesStopAfterCopy(t *testing.T) {
 	})
 }
 
+// TestPlayerCopyTablesGIPK tests the flow when the source table has a generated invisible primary key.
+// The target table needs to explicitly define the `my_row_id` column as the primary key since we will
+// be copying the rows from the source table into the target table.
+// The test also confirms that the copy_state has the gipk.
+func TestPlayerCopyTablesGIPK(t *testing.T) {
+	testVcopierTestCases(t, testPlayerCopyTablesGIPK, commonVcopierTestCases())
+}
+
+func testPlayerCopyTablesGIPK(t *testing.T) {
+	if !env.HasCapability(testenv.ServerCapabilityGeneratedInvisiblePrimaryKey) {
+		t.Skip("skipping test as server does not support generated invisible primary keys")
+	}
+	defer deleteTablet(addTablet(100))
+
+	execStatements(t, []string{
+		"SET @@session.sql_generate_invisible_primary_key=ON;",
+		"create table src1(val varbinary(128))",
+		"SET @@session.sql_generate_invisible_primary_key=OFF;",
+		"insert into src1 values('aaa'), ('bbb')",
+		fmt.Sprintf("create table %s.dst1(my_row_id int, val varbinary(128), primary key(my_row_id))", vrepldb),
```

Comment:
I understand as this is testing only, but would we want to improve support here in the future? I would imagine `my_row_id` ends up as an invisible column if we improve the support here?

Reply:
By default, the target table will continue to have the GIPK (since we copy over the table schema on the target, if the table does not already exist on the target). I updated the comments to reflect this and also enhanced the test cases to illustrate both with/without gipk.